libraries/standard/mqtt/src/mqtt.c

```diff
@@ -253,7 +280,7 @@ static int32_t sendPacket( MQTTContext_t * pContext,
     }
 
     /* Update time of last transmission if the entire packet was successfully sent. */
-    if( totalBytesSent > 0 )
+    if( ( totalBytesSent > 0 ) && ( pContext->callbacks.getTime != NULL ) )
```

Comment:
The main purpose of time function was for having keep-alive mechanism (MQTT_Processloop() API) .   MQTT_ReceiveLoop() is added to the library to address the concern raised during the design to decouple keep-alive from receive loop since some embedded applications may not want to use keep-alive and we don't want to  burden them with requirement to have time function.  So we need to consider the possibility that the time function is not provided. If getTime() is not set MQTT_ProcessLoop() API, which is a super set of MQTT_ReceiveLoop(), returns an error.

Reply:
Updated the comment